pkg/controller/statusmanager/status_manager.go

```diff
@@ -103,6 +105,9 @@ type StatusManager struct {
 
 	relatedObjects []configv1.ObjectReference
 
+	// local cache to store deployed network operator machine configs.
+	availableMachineConfigs map[string]sets.Set[string]
```

Comment:
nit: 

what do you mean with `available`? The term gives the feeling that they are already in effect. `Maybe use `rendered`?

BTW, wouldn't these machine configs be among the `relatedObjects` above as well. And an additional consideration, for pods there is no cache, they are listed every reconcile. I guess we could have done the same as an alternative although if we are filtering by owner reference it woudl be not as efficient as listing by the explicit label cNop is setting on pod related resources. I guess I don't have anything against the current approach but I am curious if you considered other options.

Reply:
`relatedObjects` would be the cache, you wouldn't need `availableMachineConfigs`.

So the thing is: 

what is our state from the time we render a MachineConfig (more-less the same time it is added to related objects) to the time we receive the pool update event (when the update our status accordingly)?

In between we have to wait to receive an event for the machine config itself. And this event can be out of order with respect to the pool event.

So you should consider:
* would it be better to update our status as soon as a machine config is rendered?
* otherwise, should we update our status  when we receive a machine config event as well?

Reply:
yes @jcaamano , nice suggestion, this helps to report the status timely as soon as machine configs are rendered. as discussed, introduced `SetMachineConfigs` method in status manager, updating caches and reporting status accordingly.